src/components/storefront/CartItem/CartItem.js

```diff
@@ -1,41 +1,42 @@
 import * as React from 'react';
+import { useEffect } from 'react';
 import { Heading, Flex, Image, useNumberInput, IconButton, VStack } from '@chakra-ui/react';
 import QuantityPicker from '../ProductDetails/QuantityPicker';
 import { CloseIcon } from '@chakra-ui/icons';
+import { useShopify } from 'hooks/useShopify';
 
-const CartItem = ({ product }) => {
-  const { images, title, variants } = product;
-  const productSku = variants[0].sku;
-  const productImage = images[0].src;
-  const productPrice = variants[0].price;
-  const productQuantity = 1; // Fetch quantity from checkout line items. This should also be a state (to update price & total)
+const CartItem = ({ title, sku, image, price, quantity, lineItemId, checkoutId }) => {
+  const { removeLineItem, updateQuantity } = useShopify();
+  const removeFromCart = () => {
+    removeLineItem(checkoutId, lineItemId);
+  };
 
   const quantityPickerProps = useNumberInput({
     step: 1,
-    defaultValue: productQuantity,
+    defaultValue: quantity,
     min: 1,
     max: 999,
   });
 
-  const removeFromCart = () => {
-    alert('To be implemented');
-  };
+  useEffect(() => {
+    updateQuantity(lineItemId, quantityPickerProps.valueAsNumber, checkoutId);
+  }, [quantityPickerProps.valueAsNumber]);
```

Comment:
This solution seemed a bit simpler/quicker than altering the `QuantityPicker` component - let me know if there are any issues with this.